osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -326,7 +330,7 @@ private void handleTimingPoints(string line)
                 beatmap.ControlPointInfo.DifficultyPoints.Add(new DifficultyControlPoint
                 {
                     Time = time,
-                    SpeedMultiplier = speedMultiplier
+                    SpeedMultiplier = MathHelper.Clamp(speedMultiplier, 0.01, 100)

```

Comment:
I really don't like this. Let's make this 0.001 at least. A simple map I tested has 0.05, and I'm sure there are harsher maps that go even lower than that.

Reply:
Maps are already clamped to these values in stable (and in fact are clamped between 0.1 and 10.0 in standard). They display as lower or higher in the editor if changed in notepad but gameplay remains within the clamped values.

Reply:
Can you point me to the code in osu-stable that does this? I don't remember seeing such a clamping outside of the editor...

Reply:
Oh, controlpoint, I see...

Reply:
Thank fuck for that.

For reference: ![Pasted Image 1.png](https://shipusercontent.com/9d9437c9d0b4fda229706423c085204d/Pasted%20Image%201.png)